src/idl2json/src/candid_types.rs

```diff
@@ -9,6 +10,7 @@ use candid::{
 ///
 /// There may be an existing conversion function
 /// but I cannot see it.
+#[allow(clippy::unimplemented)]
```

Comment:
In lib.rs I set several deny rules, including denying the use of `unimplemented!()`, `panic!()` and `expect()`. This is desirable in idl2json as it makes it harder for the code to crash for any reason, regardless of the input.  Clippy should flag any code that would crash for such a reason.  So why am I making an exception and saying that unimplemented is OK in the function below?

The code below is used to get the IDLType of a rust struct.  The type is available as `TheStructName.ty()` for any rust struct that has `#[derive(Candid)]`,however that is an internal type so I need to convert.  The internal type, however, includes some types such as Var that only appear in functions defined in IDL, not in structs, AFAIK.  Handling functions is a bit more complicated than vectors, ints and so on so if I quickly fill that in it's likely that I'll get it wrong.  But if I leave it unimplemented, can't someone craft a proposal that causes rendering to crash?  No, because this function is (currently) only used to convert normal data structures (not functions) that are compiled into the nns-dapp.  This method is not applied to any external data.

Reply:
Thanks for the explanation!